### PR TITLE
feat: Support emoji category for reaction input

### PR DESCRIPTION
**Pull request by git-babel**
# Changelog
- Added support for EmojiCategory. You can now filter emojis for different messages when adding Reactions to a message.
- Added `filterEmojiCategoryIds` to `GroupChannelProvider` and `ThreadProvider`.
    - How to Use
    ```
    const filterEmojiCategoryIds = (message: SendableMessage) => {
        if (message.customType === 'emoji_category_2') return [2];

        return [1];
    }

    <GroupChannel 
      filterEmojiCategoryIds={filterEmojiCategoryIds}
    />
    ```
    - Note: You need to set your custom EmojiCategory using [Sendbird Platform API](https://sendbird.com/docs/chat/platform-api/v3/message/reactions-and-emojis/reactions-and-emojis-overview) in advance. 

# Description
## Goal
* Customer requested that they be able to show different emojis for different EmojiCategories.  
    * e.g. show different EmojiCategory when a message is sent by a bot, versus when a message is sent by non-bot users.
## Requirements & Spec
* Users can leverage a new custom property to customize the reaction menu bar for each message. This property allows filtering specific emoji categories based on conditions, such as the `custom_type` attribute of a message.
* Existing reactions from previous versions will remain visible, although they will not appear on the reaction menu bar in the updated version to ensure a seamless transition.
## Approach
* ` filterEmojiCategoryIds: (message: SendableMessageType) => EmojiCategory['id'][];` has been introduced in `GroupChannelProvider` and `ThreadProvider`, as well as the relevant UI components. This function:
    * Can be configured at the provider level and propagated via props to individual UI components.
    * Accepts a message as a parameter and returns an array of EmojiCategory['id'], dictating which emoji categories are permissible for display on the reaction menu bar.
    * Operates internally within the reaction bar component to exclude emojis whose categories are not listed in the return array.
## Open Questions
* Is `filterEmojiCategoryIds` the optimal approach for user-driven customization of this component, or should we consider introducing a `render~~~()` property for greater flexibility?
* Any other tasks to extend public interface?

### CODE REVIEW

File: src/ui/EmojiReactions/ReactionItem.tsx [hunk 1]
```diff
@@ -49,6 +55,27 @@ export default function ReactionItem({
     && (channel?.isGroupChannel() && !channel.isSuper);
 
   const handleOnClick = () => {
+    if (isFiltered && !reactedByMe) {
+      openModal({
```
Comment (AhyoungRyu) [inline src/ui/EmojiReactions/ReactionItem.tsx:59]: @git-babel If you don't mind, could you post a screenshot of this case?

File: src/utils/index.ts [hunk 2]
```diff
@@ -557,6 +557,13 @@ export const getEmojiMapAll = (emojiContainer: EmojiContainer): Map<string, Emoj
   });
   return emojiMap;
 };
+export const getEmojiListByCategoryIds = (emojiContainer: EmojiContainer, categoryIds: number[]): Array<Emoji> => {
+  if (!categoryIds) return getEmojiListAll(emojiContainer);
+
+  return emojiContainer?.emojiCategories?.filter((emojiCategory: EmojiCategory) => categoryIds.includes(emojiCategory.id))
+    .map((emojiCategory: EmojiCategory) => emojiCategory.emojis)
+    .reduce((prevArr: Array<Emoji>, currArr: Array<Emoji>) => prevArr.concat(currArr), []);
+};
```
Comment (AhyoungRyu) [inline src/utils/index.ts:566]: Probably we can simplify this like below
```
export const getEmojiListByCategoryIds = (emojiContainer: EmojiContainer, categoryIds: number[]): Array<Emoji> => {
  if (!categoryIds) return getEmojiListAll(emojiContainer);

  return emojiContainer?.emojiCategories
    ?.filter((emojiCategory: EmojiCategory) => categoryIds.includes(emojiCategory.id))
    .flatMap((emojiCategory: EmojiCategory) => emojiCategory.emojis) || [];
};
```

Comment (git-babel) [reply src/utils/index.ts]: Applied

File: src/modules/App/MobileLayout.tsx [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import './mobile.scss';
 
 import React, { useState, useEffect } from 'react';
-import type { BaseMessage } from '@sendbird/chat/message';
```
Comment (AhyoungRyu) [inline src/modules/App/MobileLayout.tsx:4]: Could you revert this change?

File: src/ui/MessageItemReactionMenu/index.tsx [hunk 4]
```diff
@@ -70,39 +83,39 @@ export function MessageEmojiMenu({
...
-                            type={IconTypes.QUESTION}
-                            fillColor={IconColors.ON_BACKGROUND_3}
-                            width="28px"
-                            height="28px"
-                          />
-                        </div>
-                      )}
-                    />
-                  </ReactionButton>
+                    <ReactionButton
```
Comment (AhyoungRyu) [inline src/ui/MessageItemReactionMenu/index.tsx:92]: The indentations here doesn't seem to be necessary. You can put this changes back too.

**Review by AhyoungRyu — APPROVED** [review]
LGTM. Just left minor comments.

File: src/ui/EmojiReactions/ReactionItem.tsx (hunk 1 above)
Comment (git-babel) [reply src/ui/EmojiReactions/ReactionItem.tsx:59]: 스크린샷 2024-08-29 오후 4 26 12

File: src/modules/App/MobileLayout.tsx (hunk 3 above)
Comment (git-babel) [reply src/modules/App/MobileLayout.tsx:4]: `BaseMessage` is used in line 57, not by this PR. `UserMessage` is removed.

File: src/ui/MessageItemReactionMenu/index.tsx (hunk 4 above)
Comment (git-babel) [reply src/ui/MessageItemReactionMenu/index.tsx]: Applied
